devices/SparkfunLCD/SparkfunLCD.cs

```diff
@@ -0,0 +1,763 @@
+        public void TextFlowDirectionState(bool state = true)
+        {
+            if (state)
+            {
+                this.displayMode |= LCDENTRYLEFT;
+            }
+            else
+            {
+                byte tmp = LCDENTRYLEFT;
+                this.displayMode &= (byte)~tmp;
+            }
+
+            this.SpecialCommand((byte)(LCDENTRYMODESET | this.displayMode));
+        }
+
+        /// <summary>
+        /// Change state of auto-scrolling, when enabled text will be right justified
+        /// </summary>
+        /// <param name="state"><c>true</c> to enable auto-scrolling, else <c>false</c> to disable</param>
+        public void AutoScrollingState(bool state)
+        {
+            if (state)
+            {
+                this.displayMode |= LCDENTRYSHIFTINCREMENT;
+            }
+            else
+            {
+                byte tmp = LCDENTRYSHIFTINCREMENT;
+                this.displayMode &= (byte)~tmp;
+            }
+
+            this.SpecialCommand((byte)(LCDENTRYMODESET | this.displayMode));
+        }
+
+        /// <summary>
+        /// Set display contrast, default value is 120
+        /// </summary>
+        /// <param name="contrastValue">new contrast value</param>
+        public void SetContrast(byte contrastValue)
+        {
+            this.Transmit(SETTINGCOMMAND);
+            this.Transmit(CONTRASTCOMMAND);
+            this.Transmit(contrastValue);
+
+            Thread.Sleep(10);
+        }
+
+        /// <summary>
+        /// Initialize the display
+        /// </summary>
+        /// <param name="displaySize">display size</param>
+        private void Init(DISPLAYSIZE displaySize = DISPLAYSIZE.SIZE20X4)
+        {
+            switch (displaySize)
+            {
+                default:
+                case DISPLAYSIZE.SIZE20X4:
+                    this.MAXROWS = 20;
+                    this.MAXCOLUMNS = 4;
+                    break;
+                case DISPLAYSIZE.SIZE16X2:
+                    this.MAXROWS = 16;
+                    this.MAXCOLUMNS = 2;
+                    break;
+            }
+
+            this.Transmit(SPECIALCOMMAND); // Send special command character
+            this.Transmit((byte)(LCDDISPLAYCONTROL | this.displayControl)); // Send the display command
+            this.Transmit(SPECIALCOMMAND); // Send special command character
+            this.Transmit((byte)(LCDENTRYMODESET | this.displayMode)); // Send the entry mode command
+            this.Transmit(SETTINGCOMMAND); // Put LCD into setting mode
+            this.Transmit(CLEARCOMMAND); // Send clear display command
+            Thread.Sleep(50);
+        }
+
+        /// <summary>
+        /// Send data to the device
+        /// </summary>
+        /// <param name="data">data to send</param>
+        private void Transmit(byte data)
+        {
+            if (this.i2cDevice != null)
```

Comment:
Additional question: does every transmit be a full I2C cycle or can multiple bytes being transmitted in a raw in a transaction?
Because it's much faster to transfer multiple bytes than having single transactions in a raw

Reply:
full cycle transactions